### PR TITLE
Weather App API

### PR DESCRIPTION
**Pull request by zilton7**
- Uses "fetch" to get the data from the API;
- Uses "then" to wait till the data gets loaded;
- Extracts and displays required data from JSON in HTML;
- Changes text color according to the received data.

Since the main goal for this project was about utilization of APIs, design is very minimal.

Thanks for reviewing my project! ;)

### CODE REVIEW

**Review by NtwaliHeritier — COMMENTED** [review]
### **STATUS: INVALID❌❌❌❌**

Hello @zilton7 
Great work on this project, however, I invalidated this PR because of the following reasons.
- It doesn't have a package.json and webpack configurations hence unable to use it.
- The readme file doesn't correspond to the project. For example: there is no src/index.js file.

Kindly make appropriate changes and ask for another review👍👌

Happy coding💻💻💻

**Review by teekaytech — CHANGES_REQUESTED** [review]
### Status: Invalid Code Review Request :x: 

__Hi team, I commend you for the great effort put into this project. However, I have to mark the project as invalid due to the following reasons:__

- [x] You did not work with Git Flow
- The `main` or `master` branch is completely missing.
- With GitFlow, you're expected to have at least three(3) branches: production <code>master</code> or `main` branch + the development <code>develop</code> branch + Pull Requests from feature branches to the <code>develop</code> (<a href="https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/git_flow_requirement.png">screenshot</a> from Pathwright)

This Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready!


* * *
**Please, do not open a new Pull Request for subsequent reviews. You should use the same Pull Request submitted for the first review (either valid or invalid) unless it is requested otherwise.**

**Always post your inquiries and questions on this PR thread as a comment. You can mention me (@teekaytech) so I can see your comments and respond.**

* * * 

**_You can reach out to me on Twitter (@ola_lere), Github(@teekaytech), or slack (@taofeekolalere )_**

Happy Coding :man_technologist:

**Comment by zilton7** [comment]
Added main branch

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Zil Norvilis** 
Good job with your first for this project :muscle:  :muscle:
The page is properly working as expected :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [x] Your page should change its appearance based on the received data
- [x] You should use async-await for the API call
- [x] In your README file, you should include the command to run your project
- [x] Keep the application logic separated from DOM manipulation tasks 
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Comment by zilton7** [comment]
>     * [x]  Your page should change its appearance based on the received data

it does that, the Text that shows temperature changes color according to  temperature  (try Alaska and Thailand)

File: src/functions.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    const locationField = document.getElementById('location-input').value;
+    const unitSelection = document.querySelector('input[name="unit"]:checked')
+      .value;
+    fetchWeatherData(locationField, unitSelection);
+  };
+
+  const unitsFieldset = document.getElementById('units');
+  unitsFieldset.onchange = () => {
+    const locationField = document.getElementById('location-input').value;
+    const unitSelection = document.querySelector('input[name="unit"]:checked')
```
Comment (whiz25) [inline src/functions.js]: - Put this code in the dom-manipulations file. Only pure js logic should live here.

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 🙏🏿 🙏🏿 
Nice work making the required changes.
Let's make some more changes before merging.
- [x] Your app does not display the weather results. Nothing happens after hitting ENTER. Try to fix that
Screenshot from 2021-05-13 10-28-04
- [x] Separate DOM and logic from `functions.js` file. Check my comment

### Martin Afani
#### Slack: whiz25

**Review by descholar-ceo — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
## :clap: :clap: :clap: Congratulations @zilton7  , your project is approved :tada: :tada: :tada: 
You have implemented all requested changes && requirements for this project to mark it as complete, it doesn't need more changes, you can go ahead and merge all of the changes on base branch,

### Optional
- [ ] You can add a loading component when we are waiting for the result to appear
- [ ] The app should get current location of the user and displays the weather of that location when first loaded

Don't forget to fill the mandatory project completion form, and see you to next project,

Happy coding :computer: :computer: :computer: 

Re-reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)

**Pull request merged**